### DISCUSSION TITLE
Streamline and centralize issue reporting guidelines

### DISCUSSION THREAD

**Discussion by Jayman2000** [godotengine/godot-proposals#4083] [General Discussions]
At the moment, there’s three different guides that explain what should be included in [engine issues](https://github.com/godotengine/godot/issues):

- [“Filing an issue on GitHub”]
- [“Reporting bugs”]
- [Godot’s issue reporting template][issue-template]

In practice, I end up having [“Reporting bugs”] open in one browser tab, and [the new issue page] open in another. Those two pages don’t include all of the same information, so I end up switching between them multiple times to make sure that I’ve added all of the information.

To make the bug reporting process easier, here’s what I propose:

1. Update [the issue template][issue-template] so that it asks for all the same information that [“Filing an issue on GitHub”] and [“Reporting bugs”] ask for.
2. Remove [“Filing an issue on GitHub”] and [“Reporting bugs”] and replace them with something that just says “Open an issue [here][the new issue page]. Be sure to follow the template!”

I’m more than happy to open pull requests if this proposal makes sense.

Edit:
I’ve opened pull requests to implement this proposal:

- godotengine/godot#58518
- godotengine/godot-docs#5644

[“Filing an issue on GitHub”]: https://docs.godotengine.org/en/stable/community/contributing/ways_to_contribute.html#filing-an-issue-on-github
[“Reporting bugs”]: https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#reporting-bugs
[issue-template]: https://github.com/godotengine/godot/blob/master/.github/ISSUE_TEMPLATE/bug_report.yml
[the new issue page]: https://github.com/godotengine/godot/issues/new?assignees=&labels=&template=bug_report.yml

**Comment by azur-wolve** [comment]
There's a problem with templates → users really will follow them?

The current GitHUB format asks for explain non-simple things with diagrams. 
Have you seen many users on the proposals repository using diagrams?

However for bugs can be a different things so..

But users tend to not complicate themselves and not to follow the guidelines.
When they see an issue they report it in the way they want, and if there's a lot of rules to follow they get bothered and just ignore them, or abort the whole process.

Not sure how doable would it be, but allow the users to make reports from the Editor (even from projects generated with Godot) could be something convenient.

**Reply by Jayman2000** [reply]
> There's a problem with templates → users really will follow them?
> […]
> But users tend to not complicate themselves and not to follow the guidelines.
> When they see an issue they report it in the way they want, and if there's a lot of rules to follow they get bothered and just ignore them, or abort the whole process.

That’s a good point. I’d argue that this proposal would increase the chances that people follow the guidelines, but perhaps guidelines aren’t the best way to make sure that certain information gets included in bug reports. Either way, this proposal will make it easier for people who do follow the guidelines.

> Not sure how doable would it be, but allow the users to make reports from the Editor (even from projects generated with Godot) could be something convenient.

It’s definitely doable. See:

- godotengine/godot#51090

**Reply by azur-wolve** [reply]
In general I would say that people is more inclined to collaborate on stuff like that if they are "forced" to fill text boxes instead, responding a certain question or whatever.
If those are the only on-screen, following a "Next →" progression, it even can get better.

**Reply by azur-wolve** [reply]
@Jayman2000 thanks for share;
#3063 it looks like the thing will work soon, only missing the Windows Build check.

If they do it well (and continue improving it) could suppose a growth in bug reporting, since many users surely discover bugs, but are too lazy, don't know use GitHUB or are too busy. 

But if they put the way in front of them + a good documentation in the docs that shows the workflow to use it effectively could change.

**Comment by Jayman2000** [comment]
I’ve opened pull requests to implement this proposal:

- godotengine/godot#58518
- godotengine/godot-docs#5644

**Comment by mhilbrunner** [comment]
FYI, the above PRs have now been merged. Do we consider that enough for this proposal? :)

**Comment by Jayman2000** [comment]
> FYI, the above PRs have now been merged. Do we consider that enough for this proposal? :)

Definitely!
